### DISCUSSION TITLE
Confused on half rotation setting for gas meter

### DISCUSSION THREAD

**Discussion by admiralroflknife** [tronikos/esphome-magnetometer-water-gas-meter#41] [Q&A] [answered]
First up, awesome project. I love you for making it, finally getting the last monitoring item out of the way for my house (gas)

but I’m confused on the half rotation deal. All the documentation says 0.125 ft3 for the half rotation, but my meter states in its datasheet (r-275) that it does 8 revolutions per cubic foot of gas. So 0.125 makes sense, but it says for half rotation, so would t it be 0.0625 then? Or am I missing something obvious.

**Comment by tronikos** [comment, marked as answer]
The half rotation term doesn't quite apply to gas meters. There is nothing that rotates in them. The magnetometer senses the diaphragm expanding and contracting. The term is from the water meters that have a rotating magnet inside them. Just stick to the 0.125 value that all gas meters seem to be.

**Reply by admiralroflknife** [reply]
Got it, yeah the manufacturer even says 8 revolutions in their nomenclature but you right that it’s a diaphragm moving. Maybe adding some language to the readme about it would be helpful. I checked and it appears accurate to compared to my meter readings at the 0.125

**Comment by TheSerapher** [comment]
I also ran across this and with my BK-G4 gas counter which states 0,01 m3 per impulse I have just set the flow rate to exactly that. The half rotation, based on the uT values captures, looks like a full impulse of the machine? Which then makes total sense to put the 0,01 m3 per half (full) rotation and just be happy with it?

I think for Germany the documentation could state that adding the flow rate on the gas meter should be sufficient to get the volume flow properly calibrated? The numbers also matched, one full rotation of the magnetic field were 0,01 m3 on the counter.
